vuestorefront/models/product_product.py

```diff
@@ -0,0 +1,53 @@
+from odoo import models
+from odoo.addons.http_routing.models.ir_http import slug
+
+
+class ProductProduct(models.Model):
+    _inherit = 'product.product'
+
+    def get_json_ld(self):
+        self.ensure_one()
+        if self.json_ld:
+            return self.json_ld
+
+        env = self.env
+        website = env['website'].get_current_website()
+        base_url = env['ir.config_parameter'].sudo().get_param('web.base.url', '')
+        if base_url and base_url[-1:] == '/':
+            base_url = base_url[:-1]
+
+        # Get list of images
+        images = list()
+        if self.image_1920:
+            images.append('%s/web/image/product.product/%s/image' % (base_url, self.id))
+
+        json_ld = {
+            "@context": "https://schema.org/",
+            "@type": "Product",
+            "name": self.display_name,
+            "image": images,
+            "offers": {
+                "@type": "Offer",
+                "url": "%s/product/%s" % (website.domain or '', slug(self)),
+                "priceCurrency": self.currency_id.name,
+                "price": self.list_price,
+                "itemCondition": "https://schema.org/NewCondition",
+                "availability": "https://schema.org/InStock",
```

Comment:
In the long term may be good to have this set from the actual product data instead of hardcoded. Now all products will be shown on the google store and similar services as `Available` even though they may not be.